src/config.c

```diff
@@ -2133,10 +2150,20 @@ static int numericParseString(standardConfig *config, sds value, const char **er
     if (config->data.numeric.flags & OCTAL_CONFIG) {
         char *endptr;
         errno = 0;
-        *res = strtoll(value, &endptr, 8);
+        *res = strtoull(value, &endptr, 8);
         if (errno == 0 && *endptr == '\0') return 1; /* No overflow or invalid characters */
     }
 
+    /* Attempt to parse as an unsigned number */
+    if (config->data.numeric.flags & UNSIGNED_CONFIG) {
+        unsigned long long ull;
+        int ok = string2ull(value, sdslen(value), &ull);
+        if (ok) {
+            *res = (long long)ull;
```

Comment:
This feels weird, we're casting this to a ll and then casting it back. I originally thought this truncated the top bit, but apparently it preserves the bit pattern. I still think we maybe should have a dedicated function for handling unsigned numerics here.